include/common.h

```diff
@@ -2,17 +2,14 @@
 #define _COMMON_H
 
 #include "ultra64.h"
+#include "hal.h"
```

Comment:
How many files _actually_ need this? It would be nice to avoid full-rebuilds on unrelated headers changing.
Also, I feel like it'd make sense to pick either `hal` or `sys`, instead of using both.

Reply:
If we put all public system functions in header files, then their contents will change rarely, certainly not more often than common_structs.h and functions.h. And it’s more convenient, you don’t have to think every time which system files to include and which not.
I agree with the renaming of hal to sys.